action/envelope_test.go

```diff
@@ -54,6 +62,82 @@ func TestEnvelope_Proto(t *testing.T) {
 	req.Equal(tsf.payload, tsf2.payload)
 }
 
+func TestEnvelope_Actions(t *testing.T) {
+	require := require.New(t)
+	candidates := state.CandidateList{}
+	putPollResult := NewPutPollResult(1, 10001, candidates)
+
+	createStake, err := NewCreateStake(uint64(10), addr2, "100", uint32(10000), true, payload, gasLimit, gasPrice)
+	require.NoError(err)
+
+	depositToStake, err := NewDepositToStake(1, 2, big.NewInt(10).String(), payload, gasLimit, gasPrice)
+	require.NoError(err)
+
+	changeCandidate, err := NewChangeCandidate(1, candidate1Name, 2, payload, gasLimit, gasPrice)
+	require.NoError(err)
+
+	unstake, err := NewUnstake(nonce, 2, payload, gasLimit, gasPrice)
+	require.NoError(err)
+
+	withdrawStake, err := NewWithdrawStake(nonce, 2, payload, gasLimit, gasPrice)
+	require.NoError(err)
+
+	restake, err := NewRestake(nonce, index, duration, autoStake, payload, gasLimit, gasPrice)
+	require.NoError(err)
+
+	transferStake, err := NewTransferStake(nonce, cand1Addr, 2, payload, gasLimit, gasPrice)
+	require.NoError(err)
+
+	candidateRegister, err := NewCandidateRegister(nonce, candidate1Name, cand1Addr, cand1Addr, cand1Addr, big.NewInt(10).String(), 91, true, payload, gasLimit, gasPrice)
+	require.NoError(err)
+
+	candidateUpdate, err := NewCandidateUpdate(nonce, candidate1Name, cand1Addr, cand1Addr, gasLimit, gasPrice)
+	require.NoError(err)
+
+	gb := GrantRewardBuilder{}
+	grantReward := gb.Build()
+
+	cb := ClaimFromRewardingFundBuilder{}
+	claimFromRewardingFund := cb.SetAmount(big.NewInt(1)).Build()
+
+	cf := DepositToRewardingFundBuilder{}
+	depositToRewardingFund := cf.SetAmount(big.NewInt(1)).Build()
+
+	tests := []actionPayload{
+		putPollResult,
+		createStake,
+		depositToStake,
+		changeCandidate,
+		unstake,
+		withdrawStake,
+		restake,
+		transferStake,
+		candidateRegister,
+		candidateUpdate,
+		&grantReward,
+		&claimFromRewardingFund,
+		&depositToRewardingFund,
+	}
+
+	for _, test := range tests {
+		bd := &EnvelopeBuilder{}
+		elp := bd.SetNonce(1).
+			SetAction(test).
+			SetGasLimit(100000).
+			Build()
+		evlp, ok := elp.(*envelope)
+		require.True(ok)
+		err = evlp.LoadProto(evlp.Proto())
+
+		require.NoError(err)
+		require.Equal(elp.Version(), evlp.Version())
```

Comment:
The input data in https://github.com/iotexproject/iotex-core/blob/6b9ce931385dcd188de07a45ba989c3ac5430c3b/action/envelope_test.go#L70-L104 should be checked

Reply:
I don't think it needs to be checked, just to improve test coverage.

https://app.codecov.io/gh/iotexproject/iotex-core/compare/2929/changes

Reply:
ok

Reply:
let's add it in another PR